graphql/resolvers.go

```diff
@@ -3976,6 +3976,20 @@ func (r *versionResolver) Status(ctx context.Context, obj *restModel.APIVersion)
 	}
 	return status, nil
 }
+func (*versionResolver) ProjectMetadata(ctx context.Context, obj *restModel.APIVersion) (*restModel.APIProjectRef, error) {
+	projectRef, err := model.FindBranchProjectRef(*obj.Project)
```

Comment:
What parts of the project ref do you need? You probably need to use FindMergedProjectRef but I'm not sure what the use case is here.

Reply:
Is FindMergedProjectRef preferable?

Reply:
Well if you want all fields to be accurate then yes, FindMergedProjectRef is preferable. However, it is more expensive, and since most fields in the project ref are private to admins (and such they shouldn't be exposed to the general user) I'm not sure it's worth adding this complexity. Returning all fields from FindBranchProjectRef however makes it seem like those fields are populated correctly, and are allowed to be used by the version page, which isn't true, so I'm hesitant about returning all fields.

Reply:
We already return a project ref under the[ projects() resolver](https://github.com/khelif96/evergreen/blob/88c92b4ca2a08c658d760297b20881413756e0e1/graphql/resolvers.go#L1584) which is used by the project selector. Maybe we should come up with another way to block access to sensitive fields?

Reply:
Ah, that's unfortunate. That one definitely needs merging because of how it groups projects though, which I assumed was also used for the project page sidebar (although I'm not entirely sure).

Regardless then, this either should use FindMerged or just return owner/branch.

Reply:
https://jira.mongodb.org/browse/EVG-16859 I opened this to address the sensitive fields.

This field is resolved in parallel with other resolvers so i didn't notice an overall hit when using FindMergedProjectRef vs FindBranchProjectRef